Backend/app/rider-platform/rider-app/Main/src/SharedLogic/Confirm.hs

```diff
@@ -164,6 +175,24 @@ confirm DConfirmReq {..} = do
       DQuote.DriverOfferDetails driverOffer -> driverOffer.driverId
       _ -> Nothing
 
+    checkRentalBookingsOverlapping startTime rentalDuration = do
+      let statusObj =
+            DRB.BookingStatusObj
+              { normalBooking = [],
+                rentalBooking = [DRB.NEW, DRB.CONFIRMED, DRB.AWAITING_REASSIGNMENT, DRB.REALLOCATED, DRB.TRIP_ASSIGNED]
+              }
+      rentalBookings <- QRideB.findByRiderIdAndStatusObj personId statusObj
```

Comment:
This is only on rider basis. Should we check on driver basis too? They won't get `ACTIVE_BOOKING_PRESENT` too

Reply:
@0utkarsh this is overlapping checking. 
In current flow customer can book multiple rental rides, and we check here for overlapping in time
Driver can assign only one rental ride, 15 minutes before ride start, and no need to check overlapping